mergin/merginproject.py

```diff
@@ -154,10 +158,22 @@ def workspace_name(self) -> str:
         return full_name[:slash_index]
 
     def project_id(self) -> str:
-        """Returns ID of the project (UUID using 8-4-4-4-12 formatting without braces)"""
+        """Returns ID of the project (UUID using 8-4-4-4-12 formatting without braces)
+
+        Raises ClientError if project id is not present in the project metadata. This should
+        only happen with projects downloaded with old client, before February 2023,
+        see https://github.com/MerginMaps/mergin-py-client/pull/154
+        """
         if self._metadata is None:
             self._read_metadata()
-        return self._metadata["project_id"]
+
+        # "id" or "project_id" may not exist in projects downloaded with old client version
+        if "id" not in self._metadata and "project_id" not in self._metadata:
+            raise ClientError(
+                "The project directory has been created with an old version of the Mergin Maps client. Please delete the project directory and re-download the project."
+            )
+
+        return self._metadata["project_id"] if "/" in self._metadata["name"] else self._metadata["id"]
```

Comment:
could you please update this as well to use `self.is_old_metadata` ? Then also the "if" for exception does not need to check for `"id" not in self._metadata`

```python
if self.is_old_metadata:
    if "project_id" not in self._metadata:  raise ClientError(...)
    return self._metadata["project_id"]
else:
    return self._metadata["id"]
```

Reply:
Done.